diff --git a/iptables/Makefile.am b/iptables/Makefile.am.new
index fc834e0..428f2cd 100644
--- a/iptables/Makefile.am
+++ b/iptables/Makefile.am.new
@@ -7,28 +7,33 @@ BUILT_SOURCES =
 
 xtables_legacy_multi_SOURCES  = xtables-legacy-multi.c iptables-xml.c
 xtables_legacy_multi_CFLAGS   = ${AM_CFLAGS}
-xtables_legacy_multi_LDADD    = ../extensions/libext.a
+xtables_legacy_multi_LDADD    =
+xtables_legacy_multi_LDFLAGS  = -L${top_builddir}/extensions/ -liptext
 if ENABLE_STATIC
 xtables_legacy_multi_CFLAGS  += -DALL_INCLUSIVE
 endif
 if ENABLE_IPV4
 xtables_legacy_multi_SOURCES += iptables-standalone.c iptables.c
 xtables_legacy_multi_CFLAGS  += -DENABLE_IPV4
-xtables_legacy_multi_LDADD   += ../libiptc/libip4tc.la ../extensions/libext4.a
+xtables_legacy_multi_LDADD   += ../libiptc/libip4tc.la
+xtables_legacy_multi_LDFLAGS += -L${top_builddir}/extensions -liptext4
 endif
 if ENABLE_IPV6
 xtables_legacy_multi_SOURCES += ip6tables-standalone.c ip6tables.c
 xtables_legacy_multi_CFLAGS  += -DENABLE_IPV6
-xtables_legacy_multi_LDADD   += ../libiptc/libip6tc.la ../extensions/libext6.a
+xtables_legacy_multi_LDADD   += ../libiptc/libip6tc.la
+xtables_legacy_multi_LDFLAGS += -L${top_builddir}/extensions -liptext6
 endif
 xtables_legacy_multi_SOURCES += xshared.c iptables-restore.c iptables-save.c
-xtables_legacy_multi_LDADD   += ../libxtables/libxtables.la -lm
+xtables_legacy_multi_LDADD   += ${top_builddir}/libxtables/libxtables.la -lm
+
 
 # iptables using nf_tables api
 if ENABLE_NFTABLES
 xtables_nft_multi_SOURCES  = xtables-nft-multi.c iptables-xml.c
 xtables_nft_multi_CFLAGS   = ${AM_CFLAGS}
-xtables_nft_multi_LDADD    = ../extensions/libext.a ../extensions/libext_ebt.a
+xtables_nft_multi_LDADD    =
+xtables_compat_multi_LDFLAGS  = -liptext -liptext_ebt
 if ENABLE_STATIC
 xtables_nft_multi_CFLAGS  += -DALL_INCLUSIVE
 endif
@@ -42,9 +47,10 @@ xtables_nft_multi_SOURCES += xtables-save.c xtables-restore.c \
 				xtables-eb-standalone.c xtables-eb.c \
 				xtables-eb-translate.c \
 				xtables-translate.c
-xtables_nft_multi_LDADD   += ${libmnl_LIBS} ${libnftnl_LIBS} ${libnetfilter_conntrack_LIBS} ../extensions/libext4.a ../extensions/libext6.a ../extensions/libext_ebt.a ../extensions/libext_arpt.a
+xtables_nft_multi_LDADD   += ${libmnl_LIBS} ${libnftnl_LIBS} ${libnetfilter_conntrack_LIBS}
+xtables_nft_multi_LDFLAGS = -liptext4 -liptext6 -liptext_ebt -liptext_arpt
 xtables_nft_multi_SOURCES += xshared.c
-xtables_nft_multi_LDADD   += ../libxtables/libxtables.la -lm
+xtables_nft_multi_LDADD   += ${top_builddir}/libxtables/libxtables.la -lm
 endif
 
 sbin_PROGRAMS    = xtables-legacy-multi
